subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy

```diff
@@ -694,13 +695,60 @@ abstract class MakeGreen implements TransformAction<TransformParameters.None> {
-        failure.assertHasDescription("Execution failed for task ':a:resolve'.")
-        failure.assertHasCause("Execution failed for MakeGreen: ${file('b/build/b.jar')}.")
-        failure.assertHasCause("No service of type ${propertyType} available.")
+        if (configurationTime) {
+            failure.assertHasDescription("A problem occurred evaluating root project")
+            failure.assertHasCause("Cannot register artifact transform MakeGreen with parameters null")
+            failure.assertHasCause("Cannot use @InputArtifact annotation on property MakeGreen.getInput() of type ${typeName}. Allowed property types: java.io.File, org.gradle.api.provider.Provider.")
+        } else {
+            failure.assertHasDescription("Execution failed for task ':a:resolve'.")
+            failure.assertHasCause("Execution failed for MakeGreen: ${file('b/build/b.jar')}.")
+            failure.assertHasCause("Cannot use @InputArtifact annotation on property of type ${typeName}. Allowed property types: java.io.File, org.gradle.api.provider.Provider<org.gradle.api.file.FileSystemLocation>.")
+        }
+
+        where:
+        propertyType                                   | configurationTime
+        FileCollection                                 | true
+        new TypeToken<Provider<File>>() {}.getType()   | false
+        new TypeToken<Provider<String>>() {}.getType() | false
+    }
+
+    @Unroll
+    def "transform cannot use @InputArtifactDependencies to receive #propertyType"() {
+        settingsFile << """
+            include 'a', 'b', 'c'
+        """
+        setupBuildWithColorTransformAction()
+        buildFile << """
+
+project(':a') {
+    dependencies {
+        implementation project(':b')
+    }
+}
+
+abstract class MakeGreen implements TransformAction<TransformParameters.None> {
+    @${annotation.name}
+    abstract ${propertyType.name} getDependencies()
+    
+    void transform(TransformOutputs outputs) {
+        dependencies
+        throw new RuntimeException("broken")
+    }
+}
+"""
+
+        when:
+        fails(":a:resolve")
+
+        then:
+        failure.assertHasDescription("A problem occurred evaluating root project")
+        failure.assertHasCause("Cannot register artifact transform MakeGreen with parameters null")
+        failure.assertHasCause("Cannot use @InputArtifactDependencies annotation on property MakeGreen.getDependencies() of type ${propertyType.name}. Allowed property types: org.gradle.api.file.FileCollection.")
 
         where:
-        propertyType << [FileCollection, new TypeToken<Provider<File>>() {}.getType(), new TypeToken<Provider<String>>() {}.getType()]
+        annotation                | propertyType                                   | configurationTime
+        InputArtifactDependencies | File                                           | true
+        InputArtifactDependencies | String                                         | true
```

Comment:
`configurationTime` seems to be redundant.

Reply:
Removed.